lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -49,3 +52,277 @@ class NullObject:
+        # verify it does the right thing.
+        args = ["command", "arg1", "arg2"]
+        cwd = MockedPath()
+        process = tool._create_process_with_logger(args, cwd)
+
+        assert the_thread is not None
+        assert the_thread.target is log_pipe_output
+        assert the_thread.args == (subprocess.PIPE, "parent.child(logger)")
+        assert the_thread.daemon is True
+        assert the_thread.start_called is True
+
+        assert process.args is args
+        assert process.cwd is cwd
+        assert process.stdin == subprocess.DEVNULL
+        assert process.stdout == subprocess.PIPE
+        assert process.stderr == subprocess.STDOUT
+
+        # And we do it gain with a "ctx" argument value.
+        process = tool._create_process_with_logger(args, cwd, ctx="foo")
+        assert the_thread.args == (subprocess.PIPE, "parent.child(logger-foo)")
+
+    @staticmethod
+    def test_wait_for_process(caplog):
+        # First we need a Tool() object, but with a real logger.
+        mocked_install_dir = MockedPath()
+        a_logger = logging.getLogger("test_wait_for_process")
+        tool = Tool("test-tool", "opt1;opt2", mocked_install_dir, a_logger)
+
+        # We provide a mocked process object which allows us to control when
+        # the subprocess.TimeoutExpired exceptions are raised.
+        class MockProcess:
+            def __init__(self, success_on=None):
+                # Track how main times the .wait() method is called.
+                self.wait_calls = 0
+                # On which call to the .wait() method will it return a status code.
+                self.success_on = success_on
+
+            def wait(self, timeout=None):
+                self.wait_calls += 1
+                if self.success_on is not None and self.wait_calls == self.success_on:
+                    return 42
+                raise subprocess.TimeoutExpired("command", 43)
+
+        # 1 wait, 0 log entries
+        expected_log_entries = 0
+        sts = tool._wait_for_process(MockProcess(1))
+        assert sts == 42
+        assert len(caplog.record_tuples) == expected_log_entries
+
+        # 2 waits, 1 log entry
+        expected_log_entries += 1
+        sts = tool._wait_for_process(MockProcess(2))
+        assert sts == 42
+        assert len(caplog.record_tuples) == expected_log_entries
+        assert (
+            caplog.record_tuples[0][2]
+            == "None tool test-tool has not stopped after 5 seconds"
+        )
+
+        # 7 waits, 5 log entries, and use a context name.
+        expected_log_entries += 5
+        sts = tool._wait_for_process(MockProcess(7), ctx_name="bar")
+        assert sts is None
+        assert len(caplog.record_tuples) == expected_log_entries
+        assert (
+            caplog.record_tuples[5][2]
+            == "None tool test-tool bar has not stopped after 25 seconds"
+        )
+
+    @staticmethod
+    def test_wait_for_process_with_kill(caplog):
```

Comment:
You're testing multiple cases here, leveraging the same `MockProcess`; it should be easy to break these up into separate test methods (making triage easier if anything goes wrong in the future) if you push that mock up to class or module level.